crates/sui-types/src/move_package.rs

```diff
@@ -382,6 +383,18 @@ impl MovePackage {
         )
     }
 
+    // Retrieve the module with `ModuleId` in the goven pacakge.
+    // Check is the address of the module is the same of the package
+    // and return `None` if that is not the case.
+    // All modules in a package share the address with the package.
```

Comment:
@amnn is this comment even true?

Reply:
Nope. Is this ported from elsewhere?  It's generally a good idea to explicitly indicate whether `module_id` here is a `runtime_id` or a `storage_id` and document this in the doc comment as well, then it would be clearer whether this was implemented correctly.  If it's a `storage_id`, then the implementation is correct.  If it's a `runtime_id`, it's not, and the correct implementation is to fetch the module from the map first and check whether its self address matches the `ModuleId`'s address.

Reply:
I wrote that comment :) and the function, so I may very well have written it wrong, let me go and check the callers.
Thanks!

Reply:
@amnn see if this make sense to you and changing the comment.
The only call to this is from the loader here https://github.com/MystenLabs/sui/blob/main/external-crates/move/move-vm/runtime/src/loader.rs#L1073-L1081 where it explicitly uses the `storage_id`, so I guess we are good. 
Thoughts?